### PR TITLE
Add Go modules

### PR DESCRIPTION
**Pull request by nathany**
#### What does this pull request do?

Add Go modules

#### Where should the reviewer start?

go.mod

#### How should this be manually tested?

n/a

### CODE REVIEW

**Comment by dylan-bourque** [comment]
One note here, it's usually a bad idea to opt in to modules without bumping the major version.

**Comment by nathany** [comment]
@dylan-bourque Thanks for the heads up. Do you have any pointers as to what could break? I was thinking of this being a minor release update until you chimed in.

**Comment by dylan-bourque** [comment]
It's been a while since https://gitlab.com/gifts/uuid/issues/61 so I don't recall all the details. IIRC it was a bigger problem if your repo was already v2+, but in the abstract enabling support for modules is a non-trivial change in behavior. One noticeable thing is that importers who aren't using modules will get the last commit where `go.mod` doesn't exist (so no big fixes for non-modules consumers).

In contrast, though, bumping to v2 would also mean that every importer would have to change their import path (adding `/v2` at the end) to get the update.

TBH, I don't know what the _right_ path is I just didn't want to not say anything.

**Comment by nathany** [comment]
That's confusing. So rather than getting the latest commit from master, no Go-modules users would be stuck on the current 1.4.7 until they switch? (Assuming they are using Go 1.11+)?

**Comment by dylan-bourque** [comment]
Maybe it's the other way around and non-modules importers always get `HEAD` and switching to modules would "jump back" to the tagged commit.

Either way, the confusion was a big part of the advice to bump major versions when enabling modules.

**Comment by nathany** [comment]
I'm pinging #modules on Slack for some advice. Thanks for bringing this up.

**Comment by dylan-bourque** [comment]
In was going suggest exactly that.

**Comment by nathany** [comment]
> "if you are adopting modules for the first time for a pre-existing repository or set of packages that have already been tagged v2.0.0 or higher before adopting modules, then the recommended best practice is to increment the major version when first adopting modules."

It looks like https://github.com/gofrs/uuid/issues/61 was v3.

So I think we're okay.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/fsnotify/fsnotify
```
Comment (AlekSi) [inline go.mod:1]: @nathany Was it an intentional change from https://gopkg.in/fsnotify.v1, or a mistake?

Comment (nathany) [reply go.mod:1]: An intentional change.

**Pull request merged**